### ISSUE TITLE
Fedora support broken

### ISSUE THREAD

**Issue by Wunderharke** [spantaleev/matrix-docker-ansible-deploy#1610]
I suspect a095accce7f54aafb5248b19155c8aeb79bde0a6 to broke my installation.
Im using _Feora 35_ with _ntpd_ and the task ``[matrix-base : Ensure yum packages are installed]`` fails with ``No package ntp available``.
When switching to _systemd-timesyncd_ the same error occurs.

ansible --version
```
ansible 2.9.27
  config file = /opt/matrix-docker-ansible-deploy/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.9/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.9.10 (main, Jan 17 2022, 00:00:00) [GCC 11.2.1 20210728 (Red Hat 11.2.1-1)]

```

**Comment by spantaleev** [comment]
I didn't know we even support Fedora.

You can manually set what ntp service and package to use by redefining `matrix_ntpd_package` and `matrix_ntpd_service` in your `vars.yml` file.

To debug what's going on, it might be helpful to try **temporarily** adding this at the top of `roles/matrix-base/defaults/main.yml`:

```yaml
- debug: var="ansible_os_family"

- debug: var="ansible_distribution_major_version"

- fail: msg="Stop here"
 ```

We may then be able to adjust our conditional statements for `matrix_ntpd_package` and `matrix_ntpd_service` so that they work for Fedora.

-----

I suppose other things may be breaking on Fedora as well, besides this ntp stuff.

**Comment by Wunderharke** [comment]
Yes it worked for me pretty much out of the box until I accedently removed /matrix and had to run the setup again. I use an external nginx proxy on the host to deal with firewall rules.

The issue was [Changes/NtpReplacement](https://fedoraproject.org/wiki/Changes/NtpReplacement) so I defined: `matrix_ntpd_package = ntpsec` and `matrix_ntpd_service = ntpd`.
I will do some debugging and give feedback regarding the detected os_family.


Unfortunately I run into another problem namely: 
```
fatal: [matrix.scrawn.net -> 127.0.0.1]: FAILED! => changed=false 
  before: 6ef331cf302e0094ed0225b1dedec83cbdc0e54d
  msg: Local modifications exist in repository (force=no).
```
Should I open another issue for this?

**Comment by Wunderharke** [comment]
The variables you provided unfortunately did not work but I ran `aansible -m setup localhost | grep -e ansible_os_family -e ansible_distribution_major_version`
which returned:
```
        "ansible_distribution_major_version": "34",
        "ansible_os_family": "RedHat",

```
So no error on this end. But if I am reading a095accce7f54aafb5248b19155c8aeb79bde0a6 correct then `systemd-timesyncd `should be the default for `RedHat` and Version > 7

**Comment by spantaleev** [comment]
> TASK [matrix-client-element : Ensure Element themes repository is pulled]
> fatal: [XXX -> 127.0.0.1]: FAILED! => changed=false 
>   before: 6ef331cf302e0094ed0225b1dedec83cbdc0e54d
>   msg: Local modifications exist in repository (force=no).

This indicates that you've made changes to the Element themes repository that's cloned in `roles/matrix-client-element/files/scratchpad/themes`. You shouldn't be making changes there. You can go to that directory and run `git status` to see what you've changed. To fix things up, delete the whole `roles/matrix-client-element/files/scratchpad/themes` directory and the playbook will regenerate it.

-------

> So no error on this end. But if I am reading https://github.com/spantaleev/matrix-docker-ansible-deploy/commit/a095accce7f54aafb5248b19155c8aeb79bde0a6 correct then systemd-timesyncd should be the default for RedHat and Version > 7

If you're getting:

```
"ansible_distribution_major_version": "34",
"ansible_os_family": "RedHat",
```

.. then I also expect that the conditional statement we have since https://github.com/spantaleev/matrix-docker-ansible-deploy/commit/a095accce7f54aafb5248b19155c8aeb79bde0a6 will indeed make it use `systemd-timesyncd`. It's a lucky coincidence, but it's probably the right thing to do.

I expect that the `matrix-server-base` role will choose the `setup_redhat8.yml` path here: https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/e0df99a7de66e0fec8ba131605f3a39e60c51cd4/roles/matrix-base/tasks/server_base/setup.yml#L3-L7

.. which means it should be dying here for you: https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/e0df99a7de66e0fec8ba131605f3a39e60c51cd4/roles/matrix-base/tasks/server_base/setup_redhat8.yml#L27-L32

Can you modify the name of this task (e.g. `Ensure yum packages are installed` -> `Ensure yum packages are installed on RedHat 8`) and verify that it really dies at this task?

You can also add this somewhere and see what it says:

```yaml
- debug: var="matrix_ntpd_package"

- debug: var="matrix_ntpd_service"
```

**Comment by Wunderharke** [comment]
> This indicates that you've made changes to the Element themes repository that's cloned in roles/matrix-client-element/files/scratchpad/themes. You shouldn't be making changes there. You can go to that directory and run git status to see what you've changed. To fix things up, delete the whole roles/matrix-client-element/files/scratchpad/themes directory and the playbook will regenerate it.

Indeed, build.py was modified even tho I don't remember to have tampered with this file ...
After deleting the folder and setting the correct packages the setup now runs through without problems.

> Can you modify the name of this task (e.g. Ensure yum packages are installed -> Ensure yum packages are installed on RedHat 8) and verify that it really dies at this task?

I removed the manual ntp_package definition and modified the task.
To my surprise it is running `setup_redhat.yml` even though the debug shows correct values for os_family and major_version:

```
TASK [matrix-base : debug] *****************************************************
ok: [XXX] => 
  ansible_os_family: RedHat

TASK [matrix-base : debug] *****************************************************
ok: [XXX] => 
  ansible_distribution_major_version: '34'

TASK [matrix-base : debug] *****************************************************
ok: [XXX] => 
  matrix_ntpd_package: ntp

TASK [matrix-base : debug] *****************************************************
ok: [XXX] => 
  matrix_ntpd_service: ntpd

TASK [matrix-base : Ensure yum packages are installed on CentOS 7] *************
fatal: [XXX]: FAILED! => changed=false 
  failures:
  - No package ntp available.
  msg: Failed to install some of the specified packages
  rc: 1
  results: []

```

**Comment by spantaleev** [comment]
That was helpful! I belive I've fixed this in 7330992b2098db3. See the commit description for more information on what the problem was.
